### PR TITLE
Ma ac welcoming prompt to add

### LINKED ISSUE #7: 7. Como usario, quiero ver un mensaje emergente para añadir mi primer item a la lista si esta está vacía, así ayudarme a orientarme sobre cómo funciona la app
AC:

* La vista de la lista, cuando no tiene items para mostrar, debe mostrar un mensaje (ej. un botón) para que el usuario pueda agregar un nuevo item

### PR DESCRIPTION
**Pull request by alxmcr**
## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |
|  ✓   | :hammer: Refactoring       |

## Description

As a user, I want to see a welcoming prompt to add my first item if my list is empty to help me get oriented to how the app works.

## Related Issue

Fix #7 

## Acceptance Criteria

- [x]  The list view, when there are no items to display, should show a prompt (e.g., a button) for the user to add their first item.

## Changes

- Creation `EmptyList` component.
- Refactor the `ListView` component because we need to add the `EmptyList` component and check if some list of products is empty.
- Creation `ProductList` component to organize a little code (_Additional_).

## Before

image

## After

image

### CODE REVIEW

File: src/components/EmptyList/index.js
Comment (Lukaina) [inline src/components/EmptyList/index.js]: Good job, @alxmcr and @macosta08! Just a question, it is a small detail, why was it decided that for this file was `.js` and not `.jsx` as so far have been created the other components?
It is a doubt, but this does not prevent the app to work, the code is very organized and clear. Congratulations!

**Review by Lukaina — APPROVED** [review]
Congratulations, @alxmcr and @macosta08! Good job, very organized, deductive and clean code. I have tested what you did and there are no conflicts, it works well and complies with the CA. Approved!

File: src/components/EmptyList/index.js
Comment (alxmcr) [reply src/components/EmptyList/index.js]: Thanks, @Lukaina for your observation and great words! 👍 

Well, I still don't get used to using JSX files but I already changed it. 😄

Comment (Lukaina) [reply src/components/EmptyList/index.js]: @alxmcr, Many thanks to you for the work and the clarification! Don't worry, it usually happens, it just made me curious. 😄

**Review by CrisRuedaP — APPROVED** [review]
Everything is working well. Great job!

**Review by indiragp — APPROVED** [review]
Great job! Clear and organized code.
I liked the refactoring done in the ListView component :)

File: src/components/EmptyList/index.js
Comment (indiragp) [reply src/components/EmptyList/index.js]: You can read more in the React documentation: https://reactjs.org/docs/introducing-jsx.html 
A way I like to think about it is when a component has any type of HTML inside of it, it is a jsx file. If it is pure javascript, for example, the helpers file, which contain only javascript functions, then it is a .js file. 
Other common js files is when you are using redux, the actions, reducers, and containers will be js files since they don't have any HTML in them.

File: src/components/ListView/index.jsx
Comment (gmzjuliana) [inline src/components/ListView/index.jsx]: I'd use a `<p>` here as screen readers don't read `<span>`. `<span>`'s are usually used for inline decorative elements. Here's more info on them: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/span

File: src/components/ProductList/index.jsx
Comment (gmzjuliana) [inline src/components/ProductList/index.jsx]: Curious here why you've nested multiple divs and a span, would you be up for clarifying the intention here?

**Review by gmzjuliana — CHANGES_REQUESTED** [review]
looks great! left a small comment, please review before merging :)

File: src/components/ListView/index.jsx
Comment (alxmcr) [reply src/components/ListView/index.jsx]: Great! Thanks for your observation! We'll change `<span>` for `<p>`.

File: src/components/ProductList/index.jsx
Comment (alxmcr) [reply src/components/ProductList/index.jsx]: Well, we only refactored a little the code (but we didn't change something maybe also we should have improved the code 😄 ) Thanks for your question and review! We'll improve it 👍  .

File: src/components/ListView/index.jsx
Comment (alxmcr) [reply src/components/ListView/index.jsx]: **_Changes applied!_** Please, could you check our changes again? We'll appreciate it 🙏 .

File: src/components/ProductList/index.jsx
Comment (alxmcr) [reply src/components/ProductList/index.jsx]: **_Changes applied!_** 😄

File: src/components/EmptyList/index.jsx
Comment (alejandronanez) [inline src/components/EmptyList/index.jsx]: What's the reason for not using an [anchor element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a) instead?

Unless you have a strong reason to use a button that behaves like an anchor, you should use an anchor.

File: src/components/ProductList/index.jsx
Comment (alejandronanez) [inline src/components/ProductList/index.jsx]: Let's consider a more idiomatic way of doing this kind of components (where you either render some JSX or nothing).

Moving forward, I suggest you using ternary operators:

```jsx
function TheComponent() {
  return values ? <div>...</div> : null
}
```

With that, you don't have to use the `<>` anymore.

File: src/components/ProductList/index.jsx
Comment (alejandronanez) [inline src/components/ProductList/index.jsx]: I like the component name, nice work!

Let's rename this prop to something more meaningful instead as `values` is just too generic. In the end, _anything_ is a value 😅. Maybe as this component is named `ProductList` that prop could be renamed to `products`?

File: src/components/ListView/index.jsx
Comment (alejandronanez) [inline src/components/ListView/index.jsx]: I like how you're using ternaries, nice job.

**Review by alejandronanez — CHANGES_REQUESTED** [review]
👋 thanks for waiting on my review, I left some for y'all :)

The only one I consider blocking is the usage of the `button` instead of an `anchor` or a `Link` in case of React Router.

I consider the other comments important, but not merge blocking.

File: src/components/EmptyList/index.jsx
Comment (alxmcr) [reply src/components/EmptyList/index.jsx]: Well, we used a button because there was a suggestion in our acceptance criteria about a button. But, we changed `<button> `to `<Link>` of React Router, and surprise! . It's more simple and less code 😄 . Thanks @alejandronanez ! great observation 👍

File: src/components/ProductList/index.jsx
Comment (alxmcr) [reply src/components/ProductList/index.jsx]: Oh! It's awesome! It's easier! 👍 Thanks!

File: src/components/ProductList/index.jsx
Comment (alxmcr) [reply src/components/ProductList/index.jsx]: Yes, you're right! It's easier to understand `product` versus `values`. What's values? 😄  ... thanks!

**Review by gmzjuliana — APPROVED** [review]
excellent work!! looks amazing!

File: src/components/ListView/index.jsx
Comment (alxmcr) [reply src/components/ListView/index.jsx]: Well! All your suggestions to change were changed! Thanks!

**Pull request merged**
